### PR TITLE
Emit metric points as unique ETW events

### LINKED ISSUE #104: Etw-Metrics - Improvements required
Tracking item for improvements to the Etw-Metrics exporter:
https://github.com/open-telemetry/opentelemetry-rust-contrib/tree/main/opentelemetry-etw-metrics

1. Avoid heap allocation. It currently [uses a vec!](https://github.com/open-telemetry/opentelemetry-rust-contrib/blob/main/opentelemetry-etw-metrics/src/exporter/mod.rs#L68) for each export. We could re-use the vec!, or use array (allocate maximum etw size - 65360).
2. It currently serialized the entire batch of metrics and sends them as one etw event. This makes it very easy to hit the 65360 limit on ETW. We need to send one metric at a time to ETW. This is how [OTel .NET exports metrics.](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs#L224). This should significantly reduce the probability of hitting ETW size limits.
3. Unit testing - it should be possible to write unit-tests where an ETW listener is setup and it can validate the metrics are flowing as expected.

### PR DESCRIPTION
**Pull request by mattbodd**
Part of #104

## Changes
* Constructs `opentelemetry_sdk::metrics::data::ResourceMetrics` for each `opentelemetry_sdk::metrics::data::Metric` data point and emits these as ETW events

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust-contrib/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-etw-metrics/src/exporter/mod.rs [hunk 1]
```diff
@@ -63,32 +66,278 @@ impl Debug for MetricsExporter {
...
+                        resource_metrics.push(resource_metric);
+                    }
+                } else {
+                    global::handle_error(MetricsError::Other(format!(
+                        "Unsupported aggregation type: {:?}",
+                        data
+                    )));
+                }
+
+                for resource_metric in resource_metrics {
```
Comment (cijothomas) [inline opentelemetry-etw-metrics/src/exporter/mod.rs:311]: the approach used in this PR is extremely inefficient, as it clones the entire metrics and recreates them into a format that the opentelemetry-proto is able to serialize. This is okay for alpha, and will help resolve the size limit issue, but this must be done in a efficient way before stabilizing the component. (that might require support from upstream sdk)

Comment (mattbodd) [reply opentelemetry-etw-metrics/src/exporter/mod.rs:311]: Agreed. I'm working on a more performant solution but am proposing this PR as an immediate fix for #104

Comment (cijothomas) [reply opentelemetry-etw-metrics/src/exporter/mod.rs:311]: This is okay for an immediate fix.

**Review by cijothomas — APPROVED** [review]
LGTM as a mitigation for the ETW size limit issue. Definitely need to redesign this to avoid the clone/copy works.
Please add an entry to changelog.md file and we can merge.

**Pull request merged**